plugins/quickicon/eos/src/Extension/Eos.php

```diff
@@ -0,0 +1,382 @@
+<?php
+
+/**
+ * @package        Joomla.Plugin
+ * @subpackage     quickicon.eos
+ *
+ * @copyright      (C) 2023 Open Source Matters, Inc. <https://www.joomla.org>
+ * @license        GNU General Public License version 2 or later; see LICENSE.txt
+ */
+
+namespace Joomla\Plugin\Quickicon\Eos\Extension;
+
+// phpcs:disable PSR1.Files.SideEffects
+\defined('_JEXEC') or die;
+// phpcs:enable PSR1.Files.SideEffects
+
+use Exception;
+use Joomla\CMS\Access\Exception\NotAllowed;
+use Joomla\CMS\Cache\CacheController;
+use Joomla\CMS\Document\Document;
+use Joomla\CMS\Factory;
+use Joomla\CMS\HTML\HTMLHelper;
+use Joomla\CMS\Language\Text;
+use Joomla\CMS\Plugin\CMSPlugin;
+use Joomla\Database\DatabaseAwareTrait;
+use Joomla\Event\DispatcherInterface;
+use Joomla\Event\SubscriberInterface;
+use Joomla\Module\Quickicon\Administrator\Event\QuickIconsEvent;
+
+use function defined;
+
+/**
+ * Joomla! end of support notification plugin
+ *
+ * @since __DEPLOY_VERSION__
+ */
+final class Eos extends CMSPlugin implements SubscriberInterface
+{
+    use DatabaseAwareTrait;
+
+    /**
+     * The EOS date for 4.4. and beyond
+     *
+     * @var    string
+     * @since __DEPLOY_VERSION__
+     */
+    public const EOS_DATE = '2023-10-25';
+
+    /**
+     * Load the language file on instantiation.
+     *
+     * @var    bool
+     * @since __DEPLOY_VERSION__
+     */
+    protected $autoloadLanguage = true;
+
+    /**
+     * Holding the current valid message to be shown
+     *
+     * @var    array
+     * @since __DEPLOY_VERSION__
+     */
+    private $currentMessage = [];
+
+    /**
+     * The document.
+     *
+     * @var Document
+     *
+     * @since __DEPLOY_VERSION__
+     */
+    private Document $document;
+
+    /**
+     * Returns an array of events this subscriber will listen to.
+     *
+     * @return  array
+     *
+     * @since __DEPLOY_VERSION__
+     */
+    public static function getSubscribedEvents(): array
+    {
+        return [
+            'onGetIcons' => 'getEndOfServiceNotification',
```

Comment:
```suggestion
            'onGetIcons' => 'getEndOfServiceNotification',
            'onAjaxEos'  => 'onAjaxEos',
```
`onAjaxEos` event necessary